python_modules/libraries/dagster-aws/dagster_aws/sfn/sfn_launcher.py

```diff
@@ -0,0 +1,145 @@
+import logging
+import os
+from functools import cached_property
+from typing import TYPE_CHECKING, Optional
+
+import boto3
+from botocore.exceptions import ClientError
+from dagster import (
+    Field,
+    StringSource,
+    _check as check,
+)
+from dagster._core.instance import T_DagsterInstance
+from dagster._core.launcher.base import (
+    CheckRunHealthResult,
+    LaunchRunContext,
+    RunLauncher,
+    WorkerStatus,
+)
+from dagster._core.storage.dagster_run import DagsterRun
+from dagster._serdes import ConfigurableClass
+from dagster._utils.backoff import backoff
+from mypy_boto3_stepfunctions.type_defs import StartExecutionInputRequestTypeDef
+
+if TYPE_CHECKING:
+    from dagster._serdes.config_class import ConfigurableClassData
+    from mypy_boto3_stepfunctions import SFNClient
+    from mypy_boto3_stepfunctions.type_defs import (
+        DescribeExecutionOutputTypeDef,
+        StopExecutionOutputTypeDef,
+    )
+
+
+DEFAULT_RUN_TASK_RETRIES = 3
+
+
+class SFNLauncher(RunLauncher[T_DagsterInstance], ConfigurableClass):
+    def __init__(
+        self,
+        inst_data: Optional["ConfigurableClassData"],
+        sfn_arn: str,
+        name: Optional[str] = None,
+        input: Optional[str] = None,  # noqa: A002
+        trace_header: Optional[str] = None,
+    ) -> None:
+        self._client: SFNClient = boto3.client("stepfunctions")
+        self._inst_data = inst_data
+        self._sfn_arn = check.str_param(sfn_arn, "sfn_arn")
+        self._name = check.opt_str_param(name, "name") if name is not None else None
+        self._input = check.opt_str_param(input, "input") if name is not None else None
+        self._trace_header = (
+            check.opt_str_param(trace_header, "trace_header") if trace_header is not None else None
+        )
+
+    @cached_property
+    def _start_execution_input(self) -> StartExecutionInputRequestTypeDef:
+        input_dict: StartExecutionInputRequestTypeDef = {"stateMachineArn": self._sfn_arn}
+        if self._name is not None:
+            input_dict["name"] = self._name
+        if self._input is not None:
+            input_dict["input"] = self._input
+        if self._trace_header is not None:
+            input_dict["traceHeader"] = self._trace_header
+        return input_dict
+
+    def launch_run(self, context: LaunchRunContext) -> None:
+        try:
+            self._execution_arn = self._client.start_execution(**self._start_execution_input)[
+                "executionArn"
+            ]
```

Comment:
A `RunLauncher` in Dagster is responsible for kicking off the process that will orchestrate a specific `DagsterRun`.  For example the `K8sRunLauncher`  creates a kubernetes job entity with a container containing the target of the run (definitions including assets / jobs) and sets the command to a dagster cli entry point that knows how to orchestrate a run.

https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-k8s/dagster_k8s/launcher.py#L283-L295
https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_grpc/types.py#L94-L142

This implementation appears to just target a fixed sfn arn and start the execution of that, which i do not understand how it could accomplish executing the requested `DagsterRun`.